mantis-publish/mantis-publish-core/src/main/java/io/mantisrx/publish/AbstractSubscriptionTracker.java

```diff
@@ -174,34 +176,7 @@ public void refreshSubscriptions() {
         }
     }
 
-    public Optional<MantisServerSubscriptionEnvelope> getCurrentSubs(String stream) {
-        return Optional
-                .ofNullable(previousSubscriptions.get(stream))
-                .map(StreamSubscriptions::getSubsEnvelope);
-    }
-
-    static class StreamSubscriptions {
-
-        private final String streamName;
-        private final MantisServerSubscriptionEnvelope subsEnvelope;
-        private transient final long createTimeMs;
-
-        public StreamSubscriptions(final String streamName, final MantisServerSubscriptionEnvelope subsEnvelope) {
-            this.streamName = streamName;
-            this.subsEnvelope = subsEnvelope;
-            this.createTimeMs = System.currentTimeMillis();
-        }
-
-        public String getStreamName() {
-            return streamName;
-        }
-
-        public MantisServerSubscriptionEnvelope getSubsEnvelope() {
-            return subsEnvelope;
-        }
-
-        public long getCreateTimeMs() {
-            return createTimeMs;
-        }
+    protected Set<String> getCurrentSubIds(String streamName) {
+        return streamManager.getStreamSubscriptions(streamName).stream().map(Subscription::getSubscriptionId).collect(Collectors.toSet());
```

Comment:
nit: looks > 80 chars. do you want to maybe divide the chain along the operator boundary?